process/smartContract/builtInFunctions/esdtRoles.go

```diff
@@ -108,14 +108,15 @@ func getESDTRolesForAcnt(
 	acnt state.UserAccountHandler,
 	key []byte,
 ) (*esdt.ESDTRoles, bool, error) {
+	roles := &esdt.ESDTRoles{
+		Roles: make([][]byte, 0),
+	}
+
 	marshaledData, err := acnt.DataTrieTracker().RetrieveValue(key)
 	if err != nil {
-		return nil, false, err
+		return roles, true, nil
 	}
 
-	roles := &esdt.ESDTRoles{
-		Roles: make([][]byte, 0),
-	}
 	if len(marshaledData) == 0 {
```

Comment:
This condition could be moved with || in L116

Reply:
done